deploy/manifests/virtual-kubelet/base/deployment.yaml

```diff
@@ -1,17 +1,30 @@
-apiVersion: apps/v1
-kind: Deployment
+apiVersion: v1
+kind: Service
 metadata:
+  name: vk-service
   labels:
-    app: virtual-kubelet
+    app: vk-service
+spec:
+  ports:
+  - port: 80
+    name: vk-headless-service
+  clusterIP: None
+  selector:
+    app: vk-service
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
   name: virtual-kubelet
   namespace: kube-system
+  labels:
+    app: virtual-kubelet
 spec:
+  serviceName: virtual-kubelet
```

Comment:
Shouldn't this reference the service created above? It is called vk-service in the service manifest.